### PR TITLE
Add `load_json` and `save_json` to `MutliTableMetadata`

### LINKED ISSUE #891: Add load_from_json and save_to_json methods to the MultiTableMetadata
### Problem Description

_As a user, it would be useful if I could create an instance of the `MultiTableMetadata` class from a json file. It would also be useful if I could save an existing instance to a json file._

### Expected behavior

- Add a `load_from_json` method
    - Parameters:
        - `filepath`: string that is the path to json file
    - Errors:
        - If the json file doesn't contain the `SCHEMA_VERSION` attribute, then raise the following error
        ```python
        Error: This metadata file is incompatible with the SingleTableMetadata class and version.
        ``` 
    - To create the `SingleTableMetadata` instances you can call `SingleTableMetadata._load_from_dict`.

```python
>>> metadata = SingleTableMetadata.load_from_json(filepath='project/metadata.json')
```
- Add a `save_to_json` method
    - Parameters:
        - `filepath`: string that is the path to save the file to
    - Errors:
        - If there is already a file in the path location, raise the following error
        ```python
        Error: A file named 'metadata.json' already exists in this folder. Please specify a different filename.
        ```
```python
>>> metadata.save_to_json(filepath='project/metadata.json')
```

### PR DESCRIPTION
**Pull request by fealho**
Resolve #891.

### CODE REVIEW

File: sdv/metadata/utils.py [hunk 1]
```diff
@@ -150,3 +152,26 @@ def strings_from_regex(regex, max_repeat=16):
             sizes.append(size)
 
     return _from_generators(generators, max_repeat), np.prod(sizes)
+
+
+def open_path(filepath):
```
Comment (pvk-developer) [inline sdv/metadata/utils.py:157]: Instead of `open_path` maybe rename to `read_json` or similar.

**Review by pvk-developer — APPROVED** [review]
Small nit but LGTM!

File: sdv/metadata/utils.py [hunk 2]
```diff
@@ -150,3 +152,26 @@ def strings_from_regex(regex, max_repeat=16):
...
+            f"A file named '{filepath.name}' does not exist. "
+            'Please specify a different filename.'
+        )
+
+    with open(filepath, 'r', encoding='utf-8') as metadata_file:
+        return json.load(metadata_file)
+
+
+def validate_path(filepath):
+    """Validate a file path."""
```
Comment (amontanez24) [inline sdv/metadata/utils.py:171]: Can we rename this and change the description to be more descriptive like `validate_file_does_not_exist`

File: sdv/metadata/single_table.py [hunk 3]
```diff
@@ -516,19 +508,13 @@ def save_to_json(self, filepath):
...
-            )
-
+        validate_path(filepath)
         metadata = self.to_dict()
         metadata['SCHEMA_VERSION'] = self.SCHEMA_VERSION
         with open(filepath, 'w', encoding='utf-8') as metadata_file:
             json.dump(metadata, metadata_file, indent=4)
 
     def __repr__(self):
-        """Pretty print the ``SingleTableMetadata```SingleTableMetadata``."""
```
Comment (amontanez24) [inline sdv/metadata/single_table.py:532]: good catch

**Review by amontanez24 — COMMENTED** [review]
Just a small comment. After that this should be good to go!

**Review by amontanez24 — APPROVED** [review]
:shipit:

**Pull request merged**